InfiniteRecharge/src/main/java/frc/robot/subsystem/scoring/shooter/ShooterSubsystem.java

```diff
@@ -336,21 +342,46 @@ public void calculateAbsoluteDegreesToRotate() {
         }
     }
 
-    public void calculateAbsoluteDegreesElevation() {
+    // TODO: empirically test this
+        // do the distance thing here
+        // decreases as u get closer
+        // increases as u get it farther
+    public double calculateAbsoluteDegreesElevation() {
+        boolean validTarget = visionSubsystem.getValidTarget();
+        if (validTarget) {
+            double ty = visionSubsystem.getTy();
+            // double degrees = getTargetElevationDegGivenOffset(ty);
+            double distance = visionSubsystem.approximateDistanceFromTarget(ty);
+
+            List<VelocityPoint> velocityPoints = new ArrayList<VelocityPoint>();
+            // add values here
+ 
+            int closestIndex = Collections.binarySearch(velocityPoints, new VelocityPoint(distance, 0, 0));
+            // TODO: interpolate between closestIndex and our thing
+            return velocityPoints.get(closestIndex).getElevationAngle_deg();

```

Comment:
You'll want some error checking on this. Make sure closestIndex is actually a valid index in the list.